packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationFirebaseMessagingService.kt

```diff
@@ -93,14 +91,19 @@ class PushNotificationFirebaseMessagingService : FirebaseMessagingService() {
         }
     }
 
-    private fun runAppFromKilledState(){
+    private fun runAppFromKilledState(remoteMessage: RemoteMessage) {
         // Check if there is already a main Flutter Engine running
         val mainEngine = FlutterEngineCache.getInstance()
             .get(PushNotificationPluginConstants.FLUTTER_ENGINE_ID)
         if (mainEngine == null) {
-            // Create and run the default Flutter engine only when the main one is not running
-            // This calls creates the Flutter engine and runs the Flutter App's lib/main.dart
-            engineGroup.createAndRunDefaultEngine(baseContext)
+            PushNotificationBackgroundService.enqueueWork(
+                baseContext, remoteMessage.toIntent()
+            )
+        } else {
+            val notificationPayload = processRemoteMessage(remoteMessage).asChannelMap()
+            AmplifyPushNotificationsPlugin.flutterApi?.onNotificationReceivedInBackground(
+                notificationPayload
+            ) {}
```

Comment:
Because there is no need to create or use the background service when the app is just in the background but not killed, so we can directly send the notification to the dart side. 

The check for a main flutter Engine ensures we only use the background service for when the app is killed off completely.